File: app/src/main/java/com/qadam/lastpuff/ui/screens/home/HomeScreen.kt
package com.qadam.lastpuff.ui.screens.home

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.qadam.lastpuff.ui.components.HugeSosButton
import com.qadam.lastpuff.ui.components.QadamCard
import com.qadam.lastpuff.ui.components.RelapseReportCard
import com.qadam.lastpuff.ui.components.StatCard
import com.qadam.lastpuff.ui.viewmodel.AppViewModel
import com.qadam.lastpuff.util.StatsCalculator
import java.util.Locale

@Composable
fun HomeScreen(
    viewModel: AppViewModel,
    onSosClick: () -> Unit,
    onEmergencySosClick: () -> Unit
) {
    val stats by viewModel.homeStats.collectAsState()
    val profile by viewModel.profile.collectAsState()
    val moneyGoal by viewModel.moneyGoal.collectAsState()
    var showRelapseDialog by remember { mutableStateOf(false) }

    if (showRelapseDialog) {
        AlertDialog(
            onDismissRequest = { showRelapseDialog = false },
            icon = { Text("💚", style = MaterialTheme.typography.headlineMedium) },
            title = { Text("Записать срыв?") },
            text = {
                Text(
                    "Счётчик дней и сэкономленные деньги пересчитаются с этого момента.\n\n" +
                        "Один срыв не отменяет весь путь — ты продолжаешь."
                )
            },
            confirmButton = {
                TextButton(onClick = {
                    showRelapseDialog = false
                    viewModel.recordRelapse()
                }) {
                    Text(
                        "Записать и начать заново",
                        color = MaterialTheme.colorScheme.tertiary
                    )
                }
            },
            dismissButton = {
                TextButton(onClick = { showRelapseDialog = false }) {
                    Text("Отмена")
                }
            }
        )
    }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
            .padding(horizontal = 20.dp, vertical = 16.dp),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        // Главный акцент — SOS-кнопка
        HugeSosButton(
            onClick = onSosClick,
            onLongPress = onEmergencySosClick
        )

        Spacer(modifier = Modifier.height(24.dp))

        stats?.let { s ->
            // Мотивация вместо нулей
            QadamCard {
                Column(modifier = Modifier.padding(20.dp)) {
                    Text(
                        text = if (s.isJustStarted) {
                            "Сегодня ты принял важное решение."
                        } else {
                            "Ты на пути к свободе."
                        },
                        style = MaterialTheme.typography.headlineSmall,
                        fontWeight = FontWeight.SemiBold
                    )
                    Spacer(modifier = Modifier.height(8.dp))
                    Text(
                        text = s.dailyLifeCard,
                        style = MaterialTheme.typography.bodyLarge,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
            }

            s.personalReason?.let { reason ->
                Spacer(modifier = Modifier.height(12.dp))
                QadamCard {
                    Column(modifier = Modifier.padding(20.dp)) {
                        Text(
                            text = "Почему нельзя сейчас?",
                            style = MaterialTheme.typography.labelLarge,
                            color = MaterialTheme.colorScheme.primary
                        )
                        Spacer(modifier = Modifier.height(6.dp))
                        Text(
                            text = reason,
                            style = MaterialTheme.typography.titleMedium,
                            fontWeight = FontWeight.Medium
                        )
                    }
                }
            }

            s.lastVictoryAgo?.let { ago ->
                Spacer(modifier = Modifier.height(12.dp))
                QadamCard {
                    Column(modifier = Modifier.padding(16.dp)) {
                        Text(
                            text = "Последняя победа",
                            style = MaterialTheme.typography.labelLarge,
                            color = MaterialTheme.colorScheme.onSurfaceVariant
                        )
                        Text(
                            text = "Ты победил тягу $ago",
                            style = MaterialTheme.typography.bodyLarge,
                            color = MaterialTheme.colorScheme.primary
                        )
                    }
                }
            }

            if (s.totalCoins > 0) {
                Spacer(modifier = Modifier.height(12.dp))
                QadamCard {
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(16.dp),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Text("Копилка побед", style = MaterialTheme.typography.titleMedium)
                        Text("💰 ${s.totalCoins}", style = MaterialTheme.typography.headlineSmall)
                    }
                }
            }

            Spacer(modifier = Modifier.height(20.dp))
            Text(
                text = "Твой прогресс",
                style = MaterialTheme.typography.titleMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
                modifier = Modifier.fillMaxWidth()
            )
            Spacer(modifier = Modifier.height(8.dp))

            QadamCard {
                Column(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(20.dp),
                    horizontalAlignment = Alignment.CenterHorizontally
                ) {
                    Text(
                        text = "${s.days}",
                        style = MaterialTheme.typography.displayLarge,
                        color = MaterialTheme.colorScheme.primary
                    )
                    Text(
                        text = when {
                            s.days == 0 && s.hours == 0L -> "только начал"
                            s.days == 1 -> "день"
                            s.days in 2..4 -> "дня"
                            else -> "дней"
                        },
                        style = MaterialTheme.typography.titleMedium
                    )
                    if (s.days > 0 || s.hours > 0) {
                        Text(
                            text = "${s.hours} ч ${s.minutes} мин без курения",
                            style = MaterialTheme.typography.bodyMedium,
                            color = MaterialTheme.colorScheme.onSurfaceVariant
                        )
                    }
                }
            }

            Spacer(modifier = Modifier.height(12.dp))

            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                StatCard(
                    title = "Не выкурено",
                    value = if (s.cigarettesNotSmoked == 0) "—" else "${s.cigarettesNotSmoked}",
                    modifier = Modifier.weight(1f),
                    subtitle = if (s.cigarettesNotSmoked > 0) "сигарет" else "скоро здесь"
                )
                StatCard(
                    title = "Сэкономлено",
                    value = StatsCalculator.formatMoney(s.moneySaved),
                    modifier = Modifier.weight(1f),
                    subtitle = profile?.currency ?: "₸"
                )
            }

            moneyGoal?.let { goal ->
                Spacer(modifier = Modifier.height(12.dp))
                QadamCard {
                    Column(modifier = Modifier.padding(16.dp)) {
                        Text("Цель: ${goal.title}", style = MaterialTheme.typography.bodyMedium)
                        Text(
                            text = String.format(
                                Locale.getDefault(),
                                "%.0f / %.0f ${profile?.currency ?: "₸"}",
                                s.moneySaved, goal.amount
                            ),
                            style = MaterialTheme.typography.titleMedium,
                            color = MaterialTheme.colorScheme.primary
                        )
                    }
                }
            }

            Spacer(modifier = Modifier.height(20.dp))
            Text(
                text = "Честность с собой",
                style = MaterialTheme.typography.titleMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
                modifier = Modifier.fillMaxWidth()
            )
            Spacer(modifier = Modifier.height(8.dp))
            RelapseReportCard(onClick = { showRelapseDialog = true })

            Spacer(modifier = Modifier.height(12.dp))
            QadamCard {
                Column(modifier = Modifier.padding(16.dp)) {
                    Text(
                        text = "\"${s.motivationalQuote}\"",
                        style = MaterialTheme.typography.bodyLarge,
                        textAlign = TextAlign.Center,
                        modifier = Modifier.fillMaxWidth()
                    )
                }
            }
        }

        Spacer(modifier = Modifier.height(24.dp))
    }
}

File: ios/QadamLastPuff/Views/Screens/HomeScreen.swift
import SwiftUI

struct HomeScreen: View {
    @ObservedObject var viewModel: AppViewModel
    let onSosClick: () -> Void
    let onEmergencySosClick: () -> Void

    var body: some View {
        ScrollView {
            VStack(spacing: 16) {
                HugeSosButton(onTap: onSosClick, onLongPress: onEmergencySosClick)

                if let stats = viewModel.homeStats {
                    homeStatsContent(stats)
                } else {
                    QadamCard {
                        VStack(alignment: .leading, spacing: 8) {
                            Text("Настрой профиль")
                                .font(.title3.bold())
                            Text("Заполни данные в onboarding — и здесь появятся дни без курения, сэкономленные деньги и мотивация.")
                                .foregroundStyle(.secondary)
                        }
                    }
                }
            }
            .padding(20)
        }
    }

    @ViewBuilder
    private func homeStatsContent(_ stats: HomeStats) -> some View {
        QadamCard {
            VStack(alignment: .leading, spacing: 8) {
                Text(stats.isJustStarted ? "Сегодня ты принял важное решение." : "Ты на пути к свободе.")
                    .font(.title3.bold())
                Text(stats.dailyLifeCard)
                    .foregroundStyle(.secondary)
            }
        }

        if let reason = stats.personalReason {
            QadamCard {
                VStack(alignment: .leading, spacing: 6) {
                    Text("Почему нельзя сейчас?")
                        .font(.subheadline.bold())
                        .foregroundStyle(QadamColors.greenPrimary)
                    Text(reason)
                        .font(.headline)
                }
            }
        }

        if let ago = stats.lastVictoryAgo {
            QadamCard {
                VStack(alignment: .leading, spacing: 4) {
                    Text("Последняя победа").font(.subheadline).foregroundStyle(.secondary)
                    Text("Ты победил тягу \(ago)").foregroundStyle(QadamColors.greenPrimary)
                }
            }
        }

        if stats.totalCoins > 0 {
            QadamCard {
                HStack {
                    Text("Копилка побед").font(.headline)
                    Spacer()
                    Text("💰 \(stats.totalCoins)").font(.title2)
                }
            }
        }

        Text("Твой прогресс")
            .font(.headline)
            .foregroundStyle(.secondary)
            .frame(maxWidth: .infinity, alignment: .leading)

        QadamCard {
            VStack(spacing: 6) {
                Text("\(stats.days)")
                    .font(.system(size: 56, weight: .bold))
                    .foregroundStyle(QadamColors.greenPrimary)
                Text(dayLabel(stats.days, hours: stats.hours))
                if stats.days > 0 || stats.hours > 0 {
                    Text("\(stats.hours) ч \(stats.minutes) мин без курения")
                        .foregroundStyle(.secondary)
                }
            }
            .frame(maxWidth: .infinity)
        }

        HStack(spacing: 12) {
            StatCard(
                title: "Не выкурено",
                value: stats.cigarettesNotSmoked == 0 ? "—" : "\(stats.cigarettesNotSmoked)",
                subtitle: stats.cigarettesNotSmoked > 0 ? "сигарет" : "скоро здесь"
            )
            StatCard(
                title: "Сэкономлено",
                value: StatsCalculator.formatMoney(stats.moneySaved),
                subtitle: viewModel.profile?.currency ?? "₸"
            )
        }

        if let goal = viewModel.moneyGoal {
            QadamCard {
                VStack(alignment: .leading, spacing: 4) {
                    Text("Цель: \(goal.title)")
                    Text(String(format: "%.0f / %.0f \(viewModel.profile?.currency ?? "₸")", stats.moneySaved, goal.amount))
                        .font(.headline)
                        .foregroundStyle(QadamColors.greenPrimary)
                }
            }
        }

        QadamCard {
            Text("\"\(stats.motivationalQuote)\"")
                .multilineTextAlignment(.center)
                .frame(maxWidth: .infinity)
        }
    }

    private func dayLabel(_ days: Int, hours: Int64) -> String {
        if days == 0 && hours == 0 { return "только начал" }
        if days == 1 { return "день" }
        if (2...4).contains(days) { return "дня" }
        return "дней"
    }
}
